client/diffstore2/any-leaf.go

```diff
@@ -1,8 +1,9 @@
 package diffstore2
 
 import (
-	"constraints"
 	"fmt"
+
+	"golang.org/x/exp/constraints"
```

Comment:
Hi, I didn't check, but this change looks weird: constraints are out of stdlib now?

Reply:
YES,  constraints are out of stdlib.

We can found the details at : https://github.com/golang/go/issues/50792, https://utcc.utoronto.ca/~cks/space/blog/programming/Go118NoConstraintsPackage

Reply:
HI @mcluseau . Is the change good to merge ?